### PR TITLE
fix button submitting form via keyboard unexpectedly

### PR DESCRIPTION
**Pull request by simonxabris**
# Pull Request

## 📖 Description

If the element is a `fast-button` that recieved `Enter` keypress, we don't need to handle submitting the form as the native `button` rendered inside the custom element will handle it.

Important question: I'm not a 100% up-to-date with FAST releases nowdays, can this be released in the `2.x` line? It would be important for us to have this working correctly.

### 🎫 Issues

#6704 

## 📑 Test Plan

Not sure? If you want me I can write some tests for this change.

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by chrisdholt** [comment]
Thanks @simonxabris - we can backport this to the v2.x package. Do you have an example I can test this against? The outer button shouldn't be receiving focus as focus should be delegated explicitly to the internal button. A stackblitz repro/example will help validate this.

File: change/@microsoft-fast-foundation-96c3bdac-c219-4eee-ae6d-a5e64869a1a1.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-96c3bdac-c219-4eee-ae6d-a5e64869a1a1.json:2]:
```suggestion
  "type": "prerelease",
```

File: change/@microsoft-fast-foundation-96c3bdac-c219-4eee-ae6d-a5e64869a1a1.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "fix button submitting form via keyboard unexpectedly",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "abris96@gmail.com",
+  "dependentChangeType": "patch"
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-96c3bdac-c219-4eee-ae6d-a5e64869a1a1.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

File: packages/web-components/fast-foundation/src/form-associated/form-associated.ts [hunk 3]
```diff
@@ -660,7 +661,10 @@ export function FormAssociated<T extends ConstructableFormAssociated>(BaseCtor:
         private _keypressHandler(e: KeyboardEvent): void {
             switch (e.key) {
                 case keyEnter:
-                    if (this.form instanceof HTMLFormElement) {
+                    if (
+                        this.form instanceof HTMLFormElement &&
+                        !(this instanceof FASTButton)
+                    ) {
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/form-associated/form-associated.ts:667]: Is it possible to add a test for this which illustrates this preventing the current behavior?

**Comment by janechu** [comment]
Closing this as there has not been activity on this for some time.

**Pull request closed without merging**